debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java

```diff
@@ -374,18 +385,28 @@ protected void handleServerHeartbeat(Event event) {
     /**
      * Handle the supplied event that signals that an out of the ordinary event that occurred on the master. It notifies the slave
      * that something happened on the master that might cause data to be in an inconsistent state.
-     * 
+     *
      * @param event the server stopped event to be processed; may not be null
      */
     protected void handleServerIncident(Event event) {
-        logger.trace("Server incident: {}", event);
+        if (event.getData() instanceof EventDataDeserializationExceptionData) {
+            EventDataDeserializationExceptionData data = event.getData();
+
+            logger.error(
+                    String.format("Exception during event data deserialization (source info: %s, partion: %s)", source, source.partition()),
+                    data.getCause()
```

Comment:
I think we need to be much more clear that this event was skipped.

Reply:
I've changed the message as suggested. It'll say:

    Error while deserializing binlog event at offset {ts_sec=1497445917, file=mysql-bin.000002, pos=3006051, server_id=223344, event=2}.
    Use the mysqlbinlog tool to view the problematic event: `mysqlbinlog --start-position=3006194 --stop-position=3006379 --verbose mysql-bin.000002

There's an option that controls what to do in case of a deserialization error: ignore it, warn about it and proceed or stop further processing. In the second case, the message above will also contain this note:

    "This exception will be ignored and the event be skipped.

Otherwise, the exception will be raised and processing stops.